### PR TITLE
Olimex stm32 p405

### PR DESCRIPTION
**Pull request by lowlander**
A new board package for a Olimex P405 board.

### CODE REVIEW

**Comment by lowlander** [comment]
@dbkinder I have a small error (some wrong pin names) in the docs,  I'll wait with pushing until your review, so I can push possible change requests in one go.

File: boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+.. figure:: img/olimex-stm32-p405.png
+     :width: 540px
+     :align: center
+     :height: 454px
+     :alt: OLIMEX-STM32-P405
+
+     OLIMEX-STM32-P405
+
+More information about the board can be found at the
```
Comment (dbkinder) [inline boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst]: Following the template, let's add the hardware section title before this paragraph:

    Hardware
    ********

    Information about the board can be found at the

File: boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+System Clock
+============
+
+OLIMEX-STM32-P405 has two external oscillators. The frequency of
+the slow clock is 32.768 kHz. The frequency of the main clock
+is 8 MHz. The processor can setup HSE to drive the master clock,
+which can be set as high as 168 MHz.
+
+Programming and Debugging
+=========================
```
Comment (dbkinder) [inline boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst]: following the template, change the underline here to ***** (for an h2 heading)

File: boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+The OLIMEX-STM32-P405 board does not include an embedded debug tool
+interface. You will need to use ST tools or an external JTAG probe.
+In the following examples a ST-Link V2 USB dongle is used.
+
+Flashing an application to STM32F4DISCOVERY
+-------------------------------------------
```
Comment (dbkinder) [inline boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst]: ... and use ====== here for an h3 heading

File: boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+   $ make BOARD=olimex_stm32_p405 flash
+
+Run a serial host program to connect with your board:
+
+.. code-block:: console
+
+   $ minicom -D /dev/ttyACM0
+
+You should see the following message:
```
Comment (dbkinder) [inline boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst]: After you get minicom running, do you need to reset the board?

File: boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
++-------+-----------------------+----------------+
+| 10    | PA4/SPI1_NSS          | PA4            |
++-------+-----------------------+----------------+
+
+
+GPIO row of pins
+
++---------+----------------------------------------------+
+| Pin     | STM32F405 Pin Functions                      |
++=======+================================================+
```
Comment (dbkinder) [inline boards/arm/olimex_stm32_p405/doc/olimex_stm32_p405.rst]: need to have the `+` column indicator on the heading/row line, align with the previous `|`

**Review by dbkinder — CHANGES_REQUESTED** [review]
a few formatting fixes on the doc

**Comment by lowlander** [comment]
@dbkinder I applied the same docu fixes to the Olimex_stm32_e407 board, I hope you don't mind. I didn't want to open another PR for just the small doc update. And E407 is almost the same as P405.

**Review by dbkinder — DISMISSED** [review]
+1 for docs (thanks for helping keep our documentation up to date!)

**Comment by lowlander** [comment]
@galak  ping

**Review by galak — APPROVED** [review]

**Pull request merged**